### ISSUE TITLE
SelectInput and SelectArrayInput don't respect options prop when inside ReferenceInput component

### ISSUE THREAD

**Issue by JackyW83** [marmelab/react-admin#4439]
**What you were expecting:**
The SelectInput and SelectArrayInput in ReferenceInput respect fullWidth, variant in options prop

**What happened instead:**
The SelectInput and SelectArrayInput in ReferenceInput  don't respect fullWidth, variant in options prop

**Steps to reproduce:**

**Related code:**

CodeSandbox:  https://codesandbox.io/s/jovial-poitras-mjg9y?fontsize=14&hidenavigation=1&theme=dark
steps to produce:
Edit any of Post, go to MISCELLANEOUS tab, The tags input still remain "filled" variant even I change to "outlined"

**Comment by JulienMattiussi** [comment]
I wrote a PR to implement `variant='outlined'` on SelectArrayInput
This will work fine: 
```js
<ReferenceArrayInput
    reference="tags"
    source="tags"
    fullWidth
    filter={{ published: true }}
>
    <SelectArrayInput variant="outlined" />
</ReferenceArrayInput>
```

Notes : 
- `fullWitdh` prop already works, but has to be set in `ReferenceArrayInput` if you are using it.
- Usage of `fullWitdh` or `variant` in `options` prop is not needed, and not recommended because it will have some unwanted glitches

**Comment by fzaninotto** [comment]
Fixed by #4511
